src/agent/src/rpc.rs

```diff
@@ -1249,6 +1253,47 @@ impl protocols::agent_ttrpc::AgentService for AgentService {
         Err(ttrpc_error!(ttrpc::Code::INTERNAL, ""))
     }
 
+    async fn get_volume_stats(
+        &self,
+        ctx: &TtrpcContext,
+        req: VolumeStatsRequest,
+    ) -> ttrpc::Result<VolumeStatsResponse> {
+        trace_rpc_call!(ctx, "get_volume_stats", req);
+        is_allowed!(req);
+
+        info!(sl!(), "get volume stats!");
+        let mut resp = VolumeStatsResponse::new();
+
+        let mut usage_vec = Vec::new();
+        let condition = VolumeCondition::new();
+
+        // to get volume capacity stats
+        match get_volume_capacity_stats(&req.volume_guest_path) {
+            Ok(u) => {
+                usage_vec.push(u);
+            }
+            Err(e) => {
+                info!(sl!(), "fail to get volume stats!");
+                return Err(ttrpc_error!(ttrpc::Code::INTERNAL, e));
+            }
+        }
+
+        // to get volume inode stats
+        match get_volume_inode_stats(&req.volume_guest_path) {
+            Ok(u) => {
+                usage_vec.push(u);
+            }
+            Err(e) => {
+                info!(sl!(), "fail to get volume inode!");
+                return Err(ttrpc_error!(ttrpc::Code::INTERNAL, e));
+            }
+        }
+
+        resp.usage = RepeatedField::from_vec(usage_vec);
+        resp.volume_condition = SingularPtrField::some(condition);
```

Comment:
Is the condition - abnormal or not populated somewhere? How about just doing a stat on the root of the volume and report if the FS is mounted and functional or not?

Reply:
updated